dashboard/src/shared/types.ts

```diff
@@ -379,7 +380,7 @@ export interface IKubeState {
 
 export interface IBasicFormParam {
   path: string;
-  type: string;
+  type: jsonSchema.JSONSchema4TypeName | jsonSchema.JSONSchema4TypeName[] | undefined;
```

Comment:
It's not clear why we are adding `undefined` to the union here? The code change only adds the possibility of it being an array of json schema types (though this isn't tested, i don't think?).

If the reason to add `undefined` here is so existing tests which don't define it pass, can we instead update the tests? Or why is it added?

Reply:
nope, it's a compile error when readying the schema that it says that the `type` may be undefined but we can do the cast to something else there.

Reply:
Sure, but it's a compilation error *because* of something we're doing. We have explicitly defined the type here as either JSONScheme4TypeName or an array of those, not null. Let me test something, the I'll reply inline on your more recent change.